### ISSUE TITLE
Feature Request: reverse proxy option to assume ResourceNotFound by default

### ISSUE THREAD

**Issue by qrli** [Azure/service-fabric-issues#613]
With ASP.NET Core, we are using Kestrel exclusively. However, the assumption of port sharing by reverse proxy requires every microservice to add extra header `X-ServiceFabric : ResourceNotFound`. This is not microservice friendly, and we are caught every so often.

I propose an option/setting to switch the default behavior.

**Comment by Mardoxx** [comment]
Use this https://github.com/Azure/service-fabric-aspnetcore/pull/7

**Comment by qrli** [comment]
@Mardoxx Thanks for the pointer. Great info.
However, that's exactly what I want to avoid. In fact I have wrote my own middleware. (the implementation your mentioned is better than mine though :+1:  ) I still have to make sure every microservice uses it correctly. And we also have Node.js microservices, which uses a different middleware.

I'd like to avoid the cross-cutting concerns to spread across all microservices. I'd like to shorten the checklist for every new microservice.

**Comment by vturecek** [comment]
@qrli Yup, makes sense, it's something we're working on, specifically decoupling services from each other and from networking concerns in general.
